src/Stryker.Core/Stryker.Core/ProjectComponents/Fsharp/FsharpFileLeaf.cs

```diff
@@ -33,5 +36,36 @@ public override void Display()
         {
             DisplayFile(this);
         }
+
+        public override bool IsMatch(FilePattern pattern, MutantSpan span)
+        {
+            var range = FromMutantSpan(span);
+
+            return pattern.MutantSpans.Any(span => RangeModule.rangeContainsRange(FromMutantSpan(span), range));
+        }
+
+        public override IEnumerable<Range> Reduce(IEnumerable<Range> spans)
```

Comment:
Actually much will be probably taken from the TextSpanHelper, eventually using the same model as is done for span matching.

Reply:
Will the TextSpanHelper be the abstraction for span and range?

Reply:
Well as of now TextSpanHelper is a set of extensions on top of TextSpan (C#) whereas RangeHelper is a set of extensions on top of Range (F#). But let me see if I can refactor it a bit, there is some common code to extract.